### PR TITLE
MGMT-18292: in order to be able to import openshift installer we need to update assisted service as it requires new hive

### PR DESCRIPTION
**Pull request by tsorya**
[MGMT-18292](https://issues.redhat.com//browse/MGMT-18292): in order to be able to import openshift installer we need to update assisted service as it requires new hive
Changing NetworkConfig in image based config objects in order to handle null cases when we marshal the file

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -312,11 +315,11 @@ replace (
 
 replace sigs.k8s.io/cluster-api-provider-ibmcloud => sigs.k8s.io/cluster-api-provider-ibmcloud v0.9.0-alpha.0.0.20240913094112-c6bcd313bce0
 
-replace github.com/openshift/assisted-service/api => github.com/openshift/assisted-service/api v0.0.0-20230831114549-1922eda29cf8
+replace github.com/openshift/assisted-service/api => github.com/tsorya/assisted-service/api v0.0.0-20240926132224-e7d6fe26c8ff
```
Comment (eranco74) [inline go.mod:318]: Let's wait for https://github.com/openshift/assisted-service/pull/6811

**Comment by andfasano** [comment]
changes in the `agent` folder look good to me. cc @pawanpinjarkar since as the oci expert for an additional round

File: pkg/asset/imagebased/configimage/installconfig.go [hunk 2]
```diff
@@ -2,15 +2,22 @@ package configimage
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
+	"os"
+	"strings"
 
+	"github.com/pkg/errors"
```
Comment (r4f4) [inline pkg/asset/imagebased/configimage/installconfig.go:10]: Can we not introduce this deprecated package in new code?

Comment (r4f4) [reply pkg/asset/imagebased/configimage/installconfig.go:10]: You can use `fmt.Errorf` with the `%w` verb as in other parts of the code for wrapping errors.

Comment (tsorya) [reply pkg/asset/imagebased/configimage/installconfig.go:10]: It was all over the code in installer so i just used it

Comment (r4f4) [reply pkg/asset/imagebased/configimage/installconfig.go:10]: We're trying to slowly replace it as we go.

**Review by pawanpinjarkar — APPROVED** [review]
/approve

File: go.mod [hunk 3]
```diff
@@ -243,7 +244,8 @@ require (
 	github.com/opencontainers/go-digest v1.0.0 // indirect
 	github.com/opencontainers/image-spec v1.1.0 // indirect
 	github.com/opencontainers/runtime-spec v1.2.0 // indirect
-	github.com/openshift/custom-resource-status v1.1.2 // indirect
+	github.com/openshift/assisted-service v1.0.10-0.20230830164851-6573b5d7021d // indirect
```
Comment (zaneb) [inline go.mod:247]: This is an unwanted dependency caused by https://github.com/openshift/assisted-service/pull/5451
We need to watch out for these when approving go.mod changes.

**Pull request merged**